tests/integration_tests/test_sklearn.py

```diff
@@ -302,6 +313,31 @@ def test_objective_error_score_invalid() -> None:
         optuna_search.fit(X)
 
 
+@pytest.mark.parametrize(
+    "param_dist, expect",
+    [
+        ({"max_depth": distributions.IntDistribution(0, 1)}, False),
+        ({"max_depth": distributions.FloatDistribution(0, 1)}, True),
```

Comment:
Are there any reason to use `FloatDistribution`? `IntDistribution(0, 0)` seems enough to cause error in all trials.
BTW, `FloatDistribution` without `step` should be not acceptable for `BruteForceSampler` (see [docs](https://optuna.readthedocs.io/en/stable/reference/samplers/generated/optuna.samplers.BruteForceSampler.html#optuna.samplers.BruteForceSampler:~:text=(objective)-,Note,-The%20defined%20search)), so what this test catch as `ValueError` seem the one from `BruteForceSampler`, not from `OptunaSearchCV`.

Reply:
@HideakiImamura 
Is it better to implement something like `DeterministicSampler` instead of using `BruteForceSampler`?